passthrough/usecase/passthrough_usecase.go

```diff
@@ -0,0 +1,95 @@
+package usecase
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	staking "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/osmosis-labs/sqs/domain/mvc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type passthroughUseCase struct {
+	poolsUseCase mvc.PoolsUsecase
+
+	bankQueryClient banktypes.QueryClient
+	stakingQueryClient staking.QueryClient
+}
+
+var _ mvc.PassthroughUsecase = &passthroughUseCase{}
+
+// NewPassThroughUsecase Creates a passthrough use case
+func NewPassThroughUsecase(grpcURI string, puc mvc.PoolsUsecase) (mvc.PassthroughUsecase, error){
+	grpcClient, err := grpc.Dial(grpcURI, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, err
+	}
+	
+	return &passthroughUseCase{
+		poolsUseCase: puc,
+
+		bankQueryClient: banktypes.NewQueryClient(grpcClient),
+		stakingQueryClient: staking.NewQueryClient(grpcClient),
```

Comment:
Suggest initializing query clients in the caller of `NewPassThroughUsecase`. This is so that we can then create mocks for these query clients for unit testing them

Reply:
We could. However, it's generally preferable to wire up all of the dependencies in the top-level component.

Additionally, fragmenting mock and non-mock constructors would add some cognitive overhead and diverge from the patterns that have been employed in the rest of the repository.

I still think it would make sense to implement this suggestion but LMK what you think